vendor/github.com/raintank/schema/metric.go

```diff
@@ -225,6 +245,10 @@ func (m *MetricDefinition) KeyBySeries(b []byte) []byte {
 	return b
 }
 
+func (m *MetricDefinition) KeyBySeriesWithTags(b []byte) []byte {
+	return []byte(m.NameWithTags())
+}
+
```

Comment:
probably better to follow the same style as KeyBySeries and use the given input slice.
note that `b = append(b, []byte(foo)...)` has recently been optimized. It doesn't allocate anymore:
https://go-review.googlesource.com/c/go/+/109517

here's the difference in action:
```
~/g/s/g/D/s/append ❯❯❯ cat append_test.go
package main

import "testing"

var a []byte
var nameWithTags = "hello"

func NameWithTags() string {
	return nameWithTags
}

func Benchmark1(b *testing.B) {
	for i := 0; i < b.N; i++ {
		a = []byte(NameWithTags())
	}
	b.Log(len(a))
}

func Benchmark2(b *testing.B) {
	for i := 0; i < b.N; i++ {
		a = a[:0]
		a = append(a, []byte(NameWithTags())...)
	}
	b.Log(len(a))
}
~/g/s/g/D/s/append ❯❯❯ go test -bench . -benchmem -v .
goos: linux
goarch: amd64
pkg: github.com/Dieterbe/sandbox/append
Benchmark1-8   	50000000	        29.1 ns/op	       8 B/op	       1 allocs/op
--- BENCH: Benchmark1-8
    append_test.go:16: 5
    append_test.go:16: 5
    append_test.go:16: 5
    append_test.go:16: 5
    append_test.go:16: 5
Benchmark2-8   	100000000	        13.1 ns/op	       0 B/op	       0 allocs/op
--- BENCH: Benchmark2-8
    append_test.go:24: 5
    append_test.go:24: 5
    append_test.go:24: 5
    append_test.go:24: 5
    append_test.go:24: 5
PASS
ok  	github.com/Dieterbe/sandbox/append	2.811s
```

of course, since currently our callers don't actually recycle slices, we won't see a real benefit, but i rather apply the best practice here now so we can reap the benefit when we do start recycling.